Yoast/wpseo-woocommerce
# Tests/Unit: use PSR-4

## Context

* Improve dev experience 

## Summary

This PR can be summarized in the following changelog entry:

* Tests/Unit: use PSR-4

## Relevant technical choices:

### Tests/Unit: make sure all files are namespaced

... including the test bootstraps.

Includes ensuring that all calls to global functions and global constants are now fully qualified.

### Tests/Unit: move mock class to doubles directory

### Tests/Unit: remove the YoastSEO trait

On the one hand, the YoastSEO trait should by its nature be in the `doubles` directory and marked as a "mock".
On the other hand, as this trait is only used by one class, it doesn't actually make much sense to have this as a separate trait instead of integral to the test.

With this in mind, I'm removing the `YoastSEO` trait from the unit test suite in favour of adding the relevant code to the `Schema_Test` class in a similar manner as already in place in the `Twitter_Test` class.

If at some point in the future it would turn out that more tests need this exact same mock, this can be revisited, but we should code for the now, not for some hypothetical unknown future.

### Tests/Unit: use PSR-4

* Move the existing unit tests to a `Unit` subdirectory and ensure subdirectory names use "ProperCase".
* Remove prefixes from the test class names which are not needed in namespaced classes (`Yoast_WooCommerce`) and where relevant, update references to those classes in the test files.
* Ensure all test class file names match the name of the test class.
* Change the Composer `autoload-dev` directive to use PSR-4 autoloading.
* Update the `phpunit.xml.dist` file to match.
* Update the `phpcs.xml.dist` file to match.

Includes fixing copy/paste/typo errors in the test class docblocks and removing a redundant `@package` tag in a class docblock.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* _N/A_ If the tests pass, we're good.
